src/picongpu/include/particles/gasProfiles/FreeFormulaImpl.hpp

```diff
@@ -51,7 +51,8 @@ struct FreeFormulaImpl : public T_ParamClass
      */
     HDINLINE float_X operator()(const DataSpace<simDim>& totalCellOffset)
     {
-        const floatD_64 cellSize_SI( precisionCast<float_64>(cellSize) * UNIT_LENGTH );
+        const float_64 unitLength = UNIT_LENGTH; // workaround to use UNIT_LENGTH on device 
```

Comment:
quoting @BenjaminW3 and @psychocoderHPC : using the assignment operator "=" instead of a constructor is a nasty C-reminiscence in C++. It can cause casts and other weirdness that they know better than me.
(Note: general rule, more important for classes but is still true for intrinsics such as your case.)